crates/bevy_render/src/camera/visible_entities.rs

```diff
@@ -24,22 +24,104 @@ impl VisibleEntities {
     }
 }
 
+/// A mask describes which rendering group an entity belongs to.
+///
+/// Cameras with this component will only render entities with a matching
+/// mask.
+///
+/// The [`Default`] instance of `RenderingMask` returns a mask that contains
+/// no groups.
+#[derive(Copy, Clone, Debug, Reflect, PartialEq, Eq, PartialOrd, Ord)]
+#[reflect(Component)]
+pub struct RenderingMask(pub u32);
+
+impl Default for RenderingMask {
+    fn default() -> Self {
+        RenderingMask(0)
+    }
+}
+
+impl RenderingMask {
+    /// Create new `RenderingMask` with the given rendering group set.
+    pub fn group(n: u8) -> Self {
+        RenderingMask::default().set_group(n)
+    }
+
+    /// Set the given group on the mask.
+    /// This may be called multiple times to allow an entity to belong
+    /// to multiple rendering groups.
+    pub fn set_group(mut self, group: u8) -> Self {
```

Comment:
`with_group`/`without_group` would follow the usual naming convention for these methods.

Reply:
Also it's probably worth checking that group is valid (less than 31? 32?).